app/main.py

```diff
@@ -1,3 +1,70 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __eq__(self, other: float) -> bool:
+        if isinstance(other, Distance):
```

Comment:
Why use NotImplemented here?
Can you simplify your code so you use less if statements